File: Snapshot_GO/app/snapshotbitcake/ly_tell_message.go
package snapshotbitcake

import (
	"Snapshot/app"
	"Snapshot/servent/message"
)

type LYTellMessage struct {
	// This weird way of passing a basic message is for class extending(this class extends BasicMessage).
	message.BasicMessage
	LYSnapshotResult *LYSnapshotResult
}

// Constructor for LYTellMessage.
func ConstructLYTellMessage(
	sender *app.ServentInfo,
	receiver *app.ServentInfo,
	lySnapshotResult *LYSnapshotResult) *LYTellMessage {
	newLYTellMessage := LYTellMessage{
		// Pass to the "super" constructor.
		BasicMessage:     *message.ConstructBasicMessage("LY_TELL", sender, receiver, ""),
		LYSnapshotResult: lySnapshotResult,
	}

	return &newLYTellMessage
}

func (lyTellMessage *LYTellMessage) SetRedColor() message.Message {
	newLYTellMessage := lyTellMessage
	newLYTellMessage.White = false

	return newLYTellMessage
}

File: src/servent/message/snapshot/LYTellMessage.java
package servent.message.snapshot;

import java.util.List;
import java.util.Set;

import app.ServentInfo;
import app.snapshot_bitcake.LYSnapshotResult;
import servent.message.BasicMessage;
import servent.message.Message;
import servent.message.MessageType;

public class LYTellMessage extends BasicMessage {

	private static final long serialVersionUID = 3116394054726162318L;

	private Integer snapshotNumber;
	private List<LYSnapshotResult> lySnapshotResult;

	private Set<Integer> idBorderSet;

	public LYTellMessage(Integer snapshotNumber , ServentInfo sender, ServentInfo receiver, List<LYSnapshotResult> lySnapshotResult, Set<Integer> idBorderSet) {
		super(MessageType.LY_TELL, sender, receiver);

		this.snapshotNumber = snapshotNumber;
		this.lySnapshotResult = lySnapshotResult;
		this.idBorderSet = idBorderSet;
	}

	public List<LYSnapshotResult> getLYSnapshotResult() {
		return lySnapshotResult;
	}

	public Set<Integer> getIdBorderSet() {
		return idBorderSet;
	}

    /*@Override
	public Message setRedColor() {
		Message toReturn = new LYTellMessage(snapshotNumber ,getMessageType(), getOriginalSenderInfo(), getReceiverInfo(),
				false, getRoute(), getMessageText(), getMessageId(), getLYSnapshotResult());
		return toReturn;
	}*/

	public int getSnapshotNumber() {
		return snapshotNumber;
	}
}
